### PR TITLE
Clean up subscription

### LINKED ISSUE #594: `cleanupSubscription` is not working
Pulsar Functions Worker cleans the subscription in Functions Worker, see https://github.com/apache/pulsar/blob/branch-2.10/pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/FunctionActioner.java#L326-L343

Function Mesh only passes the `cleanupSubscription` to the function details, but without function worker, the config is not working on Function Mesh cluster.

### PR DESCRIPTION
**Pull request by jiangpengcheng**
*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #594, #611

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation

CleanupSubscription option doesn't work, and batch source doesn't cleanup intermediate topic either

### Modifications

Use a k8s job to do cleanup works when required

### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

Check the box below.

Need to update docs? 

- [x] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [ ] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

**Review by nlu90 — COMMENTED** [review]
After the job is finished, can we delete the job from k8s?

**Comment by jiangpengcheng** [comment]
> After the job is finished, can we delete the job from k8s?

the job is deleted automatically after it is completed since `TTLSecondsAfterFinished` is set to 0

File: controllers/spec/common.go [hunk 1]
```diff
@@ -431,8 +448,61 @@ func MakeLivenessProbe(liveness *v1alpha1.Liveness) *corev1.Probe {
...
+		VersionedParams(&corev1.PodExecOptions{
+			Command:   command,
+			Container: CleanupContainerName,
+			Stdin:     false,
+			Stderr:    true,
+			Stdout:    true,
+			TTY:       false,
+		}, scheme.ParameterCodec)
+
+	fmt.Printf("query is %v", res.URL().Query())
```
Comment (freeznet) [inline controllers/spec/common.go:487]: maybe change it to `logger` with the proper logging level?

**Review by freeznet — APPROVED** [review]

File: controllers/spec/common.go (hunk 1 above)
Comment (jiangpengcheng) [reply controllers/spec/common.go]: there are for debugging purpose, removed

**Review by freeznet — APPROVED** [review]

**Pull request merged**